decidim-conferences/app/presenters/decidim/conferences/conference_stats_presenter.rb

```diff
@@ -26,7 +26,7 @@ def component_stats(conditions)
           component_manifest.stats
                             .filter(conditions)
                             .with_context(published_components)
-                            .map { |name, data| [name, data] }.flatten
+                            .map { |name, data| [component_manifest.name, name, data] }.flatten
```

Comment:
💬 I know that you didn't change this, but I don't understand the `.flatten` part. If there are more than one record it will mix all results in just one list? That would return a list of more than three values, but the line 17 expects a list of lists with 3 values. I guess that this is usually working because the query is returning only one result, so the flatten converts `[["manifest_name", "name", "data"]]` into `["manifest_name", "name", "data"]`.

I think it would be better to call `.flatten(1)` in the outer loop (line 30) to be sure that que get a list of lists. I did a test of what I would expect to happen:

```ruby
irb(main):001:0> [[[1,2,3], [1,2,3]], [[2,3,4],[4,5,6]]].flatten(1)
=> [[1, 2, 3], [1, 2, 3], [2, 3, 4], [4, 5, 6]] 
```

What do you think?

Reply:
without flatten

`[[], [], [[:proposals, :proposals_count, 7], [:proposals, :proposals_accepted, 1], [:proposals, :supports_count, 5]], [], [], [[:accountability, :results_count, 36]], [[:debates, :debates_count, 6]], [[:sortitions, :sortitions_count, 0]], [], [[:elections, :elections_count, 6]], [], [], [[:meetings, :meetings_count, 8]], [[:proposals, :endorsements_count, 14]], [], [[:surveys, :answers_count, 0]], [], [[:debates, :endorsements_count, 0]], [], [[:blogs, :posts_count, 5]], [], []]`


with flatten

`[[], [], [:proposals, :proposals_count, 7, :proposals, :proposals_accepted, 1, :proposals, :supports_count, 5], [], [], [:accountability, :results_count, 36], [:debates, :debates_count, 6], [:sortitions, :sortitions_count, 0], [], [:elections, :elections_count, 6], [], [], [:meetings, :meetings_count, 8], [:proposals, :endorsements_count, 14], [], [:surveys, :answers_count, 0], [], [:debates, :endorsements_count, 0], [], [:blogs, :posts_count, 5], [], []]`

with flatten(1)

`[[], [], [:proposals, :proposals_count, 7, :proposals, :proposals_accepted, 1, :proposals, :supports_count, 5], [], [], [:accountability, :results_count, 36], [:debates, :debates_count, 6], [:sortitions, :sortitions_count, 0], [], [:elections, :elections_count, 6], [], [], [:meetings, :meetings_count, 8], [:proposals, :endorsements_count, 14], [], [:surveys, :answers_count, 0], [], [:debates, :endorsements_count, 0], [], [:blogs, :posts_count, 5], [], []]`

Seems that for our case, both versions will return the same output. @leio10 I think that we can leave it as it is now, what do you think?

Reply:
Mmm, but that doesn't mean that it's not working? I think that the rest of the code expects tuples of 3 values. I'd expect this method to return:
```ruby
[[:proposals, :proposals_count, 7], [:proposals, :proposals_accepted, 1], [:proposals, :supports_count, 5], [:accountability, :results_count, 36], [:debates, :debates_count, 6], [:sortitions, :sortitions_count, 0], [:elections, :elections_count, 6], [:meetings, :meetings_count, 8], [:proposals, :endorsements_count, 14], [:surveys, :answers_count, 0], [:debates, :endorsements_count, 0], [:blogs, :posts_count, 5]]
```

If that's true, we can just do `.sum` instead of `.flatten(1)`. What do you think?

Reply:
In `def statistics(grouped_stats)` is generated the "final" version of the statistics. There, for each manifest name, we extract each statistic and his value. Bellow is the first iteration, for manifest_name: "proposal"

```
backend-app        | :proposals
backend-app        | [[:proposals, :proposals_count, 7, :proposals, :proposals_accepted, 1, :proposals, :supports_count, 5], [:proposals, :endorsements_count, 14]]

backend-app        | proposals
backend-app        | proposals_count
backend-app        | 7
backend-app        | proposals
backend-app        | proposals_accepted
backend-app        | 1
backend-app        | proposals
backend-app        | supports_count
backend-app        | 5
```

So, having all statistics for proposals (or other manifest) on the same list, it's ok. 

Version with "sum" will generate something like this:
```
[0,
 0,
 [:proposals,
  :proposals_count,
  7,
  :proposals,
  :proposals_accepted,
  1,
  :proposals,
  :supports_count,
  5],
 0,
 0,
 [:accountability, :results_count, 36],
 [:debates, :debates_count, 6],
 [:sortitions, :sortitions_count, 0],
 0,
 [:elections, :elections_count, 6],
 0,
 0,
 [:meetings, :meetings_count, 8],
 [:proposals, :endorsements_count, 14],
 0,
 [:surveys, :answers_count, 0],
 0,
 [:debates, :endorsements_count, 0],
 0,
 [:blogs, :posts_count, 5],
 0,
 0]
```
And for sum we will also need to change `highlighted_stats = highlighted_stats.reject { |_manifest, _name, data| data.zero? }`

Reply:
ok, sorry, let's merge it as it is 👍